### ISSUE TITLE
Realm complains computed property's type in Unit test

### ISSUE THREAD

**Issue by owenzhao** [realm/realm-cocoa#2815]
I have a computed property, type of NSURL, as far as I know that computed property is not persist. So it should not be complained.

>failed: caught "RLMException", "'NSURL' is not supported as an RLMObject property. All properties must be primitives, NSString, NSDate, NSData, RLMArray, or subclasses of RLMObject. See http://realm.io/docs/objc/api/Classes/RLMObject.html for more information."

````swift
class CFRealmBaseURL: Object {
    dynamic var path = ""
    
    override class func primaryKey() -> String? {
        return "path"
    }
}

class CFRealmProjectStatus: Object {
    dynamic var text = "一般任务" //"等待回复", "等待材料", "任务完成"
    dynamic var id = 0 //1, 2, -1
    
    override class func primaryKey() -> String? {
        return "id"
    }
}

class CFRealmCard: Object {
    dynamic var baseURL = CFRealmBaseURL()
    dynamic var projectType = ""
    dynamic var company = ""
    dynamic var projectNumber = ""
    dynamic var projectProcess = ""
    var relativeURL:NSURL {
        get {
            let path = [self.projectProcess, self.company, self.projectNumber, self.projectProcess].joinWithSeparator("/")
            return NSURL(fileURLWithPath: path, isDirectory: true, relativeToURL: NSURL(fileURLWithPath: self.baseURL.path, isDirectory: true))
        }
        
        set(newURL) {
            let paths = newURL.pathComponents!
            
            if let baseURLCount = newURL.baseURL?.pathComponents?.count {
                self.baseURL.path = newURL.baseURL!.absoluteString
                self.projectType = paths[baseURLCount]
                self.company = paths[baseURLCount + 1]
                self.projectNumber = paths[baseURLCount + 2]
                self.projectProcess = paths[Range(start: baseURLCount + 3, end: paths.endIndex)].joinWithSeparator("/")
            }
        }
    }
    
    dynamic var projectStatus = CFRealmProjectStatus()
    dynamic var projectSummary = ""
    let contactPersonList = List<CFRealmProjectContactPerson>()
    
    dynamic var createdDate = NSDate(timeIntervalSince1970: 0)
    
    func setCard(baseURLs:[NSURL], url:NSURL) {
        let fullPath = url.absoluteString
        for aBaseURL in baseURLs { //baseURLs is order by levels from deep to low
            let aBasePath = aBaseURL.absoluteString
            if fullPath.hasPrefix(aBasePath) {
                var path = fullPath
                path.removeRange(Range(start: aBasePath.startIndex, end: aBasePath.endIndex))
                self.relativeURL = NSURL(fileURLWithPath: path, isDirectory: true, relativeToURL: aBaseURL)
            }
        }
    }
````

I have to add the computed property to ignore list to stop the complain.

````swift
    override class func ignoredProperties() -> [String] {
        return ["relativeURL"]
    }
````

**Comment by jpsim** [comment]
There's no way to determine at run-time whether or not a Swift class property has the default setters & getters, or custom getters & setters. So in this case, Realm has *no other way of knowing* that your `relativeURL` property should be ignored other than including it in `ignoredProperties()`.

**Comment by tom-sparo** [comment]
@jpsim can you clarify this in the doc writeups as well?  I was lead astray by a comment in the example code to believe that Realm auto-ignored all computed properties (not just computed properties that were objective c types).

Specifically talking about this code segment:
```swift
class Person: Object {
  dynamic var tmpID = 0
  var name: String { // computed properties are automatically ignored
    return "\(firstName) \(lastName)"
  }
  dynamic var firstName = ""
  dynamic var lastName = ""

  override static func ignoredProperties() -> [String] {
    return ["tmpID"]
  }
}
```

Is that comment inaccurate/incomplete and computed Objc type properties are automatically ignored?

As an aside, the error coming out of the Realm system was a little unhelpful.  I had to put a break point in to figure out which property in particular was sending this error.

**Comment by jpsim** [comment]
I see how that could be confusing. I guess I just wouldn't refer to a property with a custom setter as a "computed property", which I only took to mean "properties whose getters return a value computed from other values".

I just updated the comment wording to say "read-only properties are automatically ignored". It should take 4-5 minutes to be updated on the live site.

Thanks for the tip @owenzhao and @tom-sparo.
